### PR TITLE
Replace `unittest` with `pytest`

### LINKED ISSUE #88: chore: Replace `TestCase` with `pytest` test suite or test case
Same as https://github.com/kytos-ng/sdntrace/issues/53

### PR DESCRIPTION
**Pull request by Alopalao**
Closes #88 

### Summary

Replace `unittest` with `pytest`

### Local Tests
Run tox

### End-to-End Tests
N/A

### CODE REVIEW

File: tests/unit/v0x01/test_symmetric/test_vendor_header.py [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
-
-class TestVendorHeader(TestStruct):
+class TestVendorHeader(StructTest):
     """Vendor message tests (also those in :class:`.TestDump`)."""
 
-    @classmethod
-    def setUpClass(cls):
-        super().set_minimum_size(8)
+    def setup_method(self):
+        self.set_raw_dump_object(Hello, xid=1)
```
Comment (Alopalao) [inline tests/unit/v0x01/test_symmetric/test_vendor_header.py:10]: Added this line here from previous file [tests/unit/v0x01/test_symmetric/test_hello.py](https://github.com/kytos-ng/python-openflow/pull/90/files#diff-a3369021f2c46c4daea3ddd4a9515f2064fa7019969e958a1e06eb0d8672fa96). I noticed that with `unittest` it was also getting values from that file.

**Comment by Alopalao** [comment]
Overall number of test have been reduced drastically from 400 to 247 but the percentage of coverage is the same. This is because every child of `tests/unit/test_utils.py ` now does not call a second time every test from its parent. Is this correct?

**Comment by viniarck** [comment]
> Overall number of test have been reduced drastically from 400 to 247 but the percentage of coverage is the same. This is because every child of `tests/unit/test_utils.py ` now does not call a second time every test from its parent. Is this correct?

```
Prev  (stmt / miss / percent) : 5990    373    94%
After (stmt / miss / percent) : 5795    375    94%
```

@Alopalao indeed the number of displayed collected test cases changed, but yes, coverage is equivalent, maybe with `unitest.TestCase` it was counting more than once in certain cases due to inheritance base class also being a test. 

I've also looked into a detailed diff comparing the execution and I noticed no relevant difference. By the way, pretty neat how `StructTest` ended up, relatively minimal refactoring in that part while still compatible.

**Review by viniarck — APPROVED** [review]
Excellent PR, Aldo. 

LGTM. This will be shipped soon.

**Pull request merged**